### PR TITLE
integration-tests/smoke: move ccip msging test

### PR DESCRIPTION
**Pull request by makramkd**
Move the CCIP messaging test from deployment/ccip to integration-tests/smoke, and transform it into a docker-based test instead of simulated backend.

### CODE REVIEW

File: .github/e2e-tests.yml [hunk 1]
```diff
@@ -948,6 +948,20 @@ runner-test-matrix:
...
+  
+  - id: smoke/ccip_messaging_test.go:*
+    path: integration-tests/smoke/ccip_messaging_test.go
+    test_env_type: docker
+    runs_on: ubuntu-latest
+    triggers:
+      - PR E2E Core Tests
+      - Merge Queue E2E Core Tests
+      - Nightly E2E Tests
+    test_cmd: cd integration-tests/ && go test smoke/ccip_messaging_test.go -timeout 12m -test.parallel=1 -count=1 -json
```
Comment (dimkouv) [inline .github/e2e-tests.yml:960]: nit:
```
--- cd integration-tests/ && go test smoke/ccip_messaging_test.go
+++ go test integration-tests/smoke/ccip_messaging_test.go
```
?

Comment (dimkouv) [reply .github/e2e-tests.yml:960]: remove `-count=1` ?

**Review by dimkouv — APPROVED** [review]

File: integration-tests/smoke/ccip_messaging_test.go [hunk 2]
```diff
@@ -34,12 +39,13 @@ type messagingTestCaseOutput struct {
 	nonce    uint64
 }
 
-func Test_Messaging(t *testing.T) {
-	t.Parallel()
```
Comment (0xnogo) [inline integration-tests/smoke/ccip_messaging_test.go:38]: Why to remove this?

Comment (makramkd) [reply integration-tests/smoke/ccip_messaging_test.go:38]: The `t.Parallel`?

Comment (0xnogo) [reply integration-tests/smoke/ccip_messaging_test.go:38]: Yes, are we parallelizing somewhere else?

Comment (makramkd) [reply integration-tests/smoke/ccip_messaging_test.go:38]: Right now this test is being run in its own GitHub job, so prolly doesn't need t.Parallel() right now

File: .github/e2e-tests.yml (hunk 1 above)
Comment (makramkd) [reply .github/e2e-tests.yml:960]: Just a copypasta from the rest of the matrix, would rather stay consistent

**Review by 0xnogo — APPROVED** [review]

**Review by asoliman92 — APPROVED** [review]

**Review by gheorghestrimtu — APPROVED** [review]

File: .github/e2e-tests.yml (hunk 1 above)
Comment (kalverra) [reply .github/e2e-tests.yml:960]: `-count=1` is important in E2E tests as it disables Go test caching. Which, in the case of these E2E tests, can lead to false negatives.

**Pull request merged**